File: modules/protocol/769/src/main/java/candle/protocol/codec/LongCodec.java
package candle.protocol.codec;

import candle.types.primitives.Int64;
import candle.types.resource.Identifier;

import java.io.*;

public class LongCodec extends TypeCodec<Int64> {
  private static TypeIdentifier<Int64> IDENTIFIER;

  public LongCodec() {
    if ( IDENTIFIER == null ) {
      IDENTIFIER = identifier("minecraft", "long");
    }
  }

  @Override
  Int64 read( InputStream in, AbstractDataCodec dataCodec ) throws
                                                            IOException {
    DataInputStream dataIn = new DataInputStream(in);
    return new Int64(dataIn.readLong());
  }

  @Override
  void write( OutputStream out, Int64 value, AbstractDataCodec dataCodec ) throws
                                                                           IOException {
    DataOutputStream dataOut = new DataOutputStream(out);
    dataOut.writeLong(value.value());
  }

  @Override
  public Identifier namespacedIdentifier() {
    return null;
  }
}

File: protocol/769/src/main/kotlin/net/candlemc/protocol/codec/LongCodec.kt
package net.candlemc.protocol.codec

import net.candlemc.types.resource.Identifier
import java.io.*

class LongCodec : TypeCodec<Long>() {
    init {
        if (IDENTIFIER == null) {
            IDENTIFIER = identifier("minecraft", "long")
        }
    }

    @Throws(IOException::class)
    override fun read(inputStream: InputStream, dataCodec: AbstractDataCodec): Long {
        val dataIn = DataInputStream(inputStream)
        return dataIn.readLong()
    }

    @Throws(IOException::class)
    override fun write(outputStream: OutputStream, value: Long, dataCodec: AbstractDataCodec) {
        val dataOut = DataOutputStream(outputStream)
        dataOut.writeLong(value)
    }

    override fun namespacedIdentifier(): Identifier {
        return IDENTIFIER!!
    }

    companion object {
        private var IDENTIFIER: TypeIdentifier<Long>? = null

        fun identifier(): TypeIdentifier<Long>? {
            return IDENTIFIER
        }
    }
}
